src/test/unit/BLSSignatureCheckerUnit.t.sol

```diff
@@ -0,0 +1,229 @@
+        );
+    }
+
+    function testBLSSignatureChecker_IncorrectQuorumAPK_Reverts(uint256 pseudoRandomNumber) public {
+        uint256 numNonSigners = pseudoRandomNumber % (maxOperatorsToRegister - 2) + 1;
+
+        uint256 quorumBitmap = 1;
+        bytes memory quorumNumbers = BitmapUtils.bitmapToBytesArray(quorumBitmap);
+
+        (uint32 referenceBlockNumber, BLSSignatureChecker.NonSignerStakesAndSignature memory nonSignerStakesAndSignature) = 
+            _registerSignatoriesAndGetNonSignerStakeAndSignatureRandom(pseudoRandomNumber, numNonSigners, quorumBitmap);
+        
+        // set the quorumApk to a different value
+        nonSignerStakesAndSignature.quorumApks[0] = nonSignerStakesAndSignature.quorumApks[0].negate();
+
+        cheats.expectRevert("BLSSignatureChecker.checkSignatures: quorumApk hash in storage does not match provided quorum apk");
+        blsSignatureChecker.checkSignatures(
+            msgHash, 
+            quorumNumbers,
+            referenceBlockNumber, 
+            nonSignerStakesAndSignature
+        );
+    }
+
+    function testBLSSignatureChecker_IncorrectSignature_Reverts(uint256 pseudoRandomNumber) public {
+        uint256 numNonSigners = pseudoRandomNumber % (maxOperatorsToRegister - 2) + 1;
+
+        uint256 quorumBitmap = 1;
+        bytes memory quorumNumbers = BitmapUtils.bitmapToBytesArray(quorumBitmap);
+
+        (uint32 referenceBlockNumber, BLSSignatureChecker.NonSignerStakesAndSignature memory nonSignerStakesAndSignature) = 
+            _registerSignatoriesAndGetNonSignerStakeAndSignatureRandom(pseudoRandomNumber, numNonSigners, quorumBitmap);
+        
+        // set the sigma to a different value
+        nonSignerStakesAndSignature.sigma = nonSignerStakesAndSignature.sigma.negate();
+
+        cheats.expectRevert("BLSSignatureChecker.checkSignatures: signature is invalid");
+        blsSignatureChecker.checkSignatures(
+            msgHash, 
+            quorumNumbers,
+            referenceBlockNumber, 
+            nonSignerStakesAndSignature
+        );
+    }
+
+    function testBLSSignatureChecker_InvalidSignature_Reverts(uint256 pseudoRandomNumber) public {
+        uint256 numNonSigners = pseudoRandomNumber % (maxOperatorsToRegister - 2) + 1;
+
+        uint256 quorumBitmap = 1;
+        bytes memory quorumNumbers = BitmapUtils.bitmapToBytesArray(quorumBitmap);
+
+        (uint32 referenceBlockNumber, BLSSignatureChecker.NonSignerStakesAndSignature memory nonSignerStakesAndSignature) = 
+            _registerSignatoriesAndGetNonSignerStakeAndSignatureRandom(pseudoRandomNumber, numNonSigners, quorumBitmap);
+        
+        // set the sigma to a different value
+        nonSignerStakesAndSignature.sigma.X++;
+
+        cheats.expectRevert();
+        blsSignatureChecker.checkSignatures(
+            msgHash, 
+            quorumNumbers,
+            referenceBlockNumber, 
+            nonSignerStakesAndSignature
+        );
+    }
```

Comment:
should this have a specific revert message or does this end up in some low-level call failure (like a failure in the precompile call?) ?

Reply:
precompile call fails rip

Reply:
OK plz add a comment like
```
        // expect a non-specific low-level revert, since this call will ultimately fail as part of the precompile call
        cheats.expectRevert();
        blsSignatureChecker.checkSignatures(
            msgHash, 
            quorumNumbers,
            referenceBlockNumber, 
            nonSignerStakesAndSignature
        );
```